src/Target.cpp

```diff
@@ -753,6 +772,78 @@ int Target::natural_vector_size(const Halide::Type &t) const {
     }
 }
 
+bool Target::get_runtime_compatible_target(const Target& other, Target &result) {
+    const std::array<Feature, 25> union_features = {
```

Comment:
(1) Some commenting about the logic here is appropriate.
(2) As new features are added, they may need to be added to one of these sets; at a minimum, we should add a comment to the declaration in HalideRuntime.h to warn people of this.